creator-node/src/services/stateMachineManager/monitoring/StateMonitoringQueue.js

```diff
@@ -0,0 +1,254 @@
+    // Add handlers for logging
+    queue.on('global:waiting', (jobId) => {
+      this.log(`Queue Job Waiting - ID ${jobId}`)
+    })
+    queue.on('global:active', (jobId, jobPromise) => {
+      this.log(`Queue Job Active - ID ${jobId}`)
+    })
+    queue.on('global:lock-extension-failed', (jobId, err) => {
+      this.logError(
+        `Queue Job Lock Extension Failed - ID ${jobId} - Error ${err}`
+      )
+    })
+    queue.on('global:stalled', (jobId) => {
+      this.logError(`stateMachineQueue Job Stalled - ID ${jobId}`)
+    })
+    queue.on('global:error', (error) => {
+      this.logError(`Queue Job Error - ${error}. Queuing another job...`)
+    })
+
+    // Re-queue a new job when the current job fails or succeeds
+    queue.on('completed', (job, result) => {
+      this.log(
+        `Queue Job Completed - ID ${job?.id} - Result ${JSON.stringify(
+          result
+        )}. Queuing another job...`
+      )
+      if (result?.jobFailed) {
+        this.enqueueJobAfterSuccess(queue, job, result)
+      } else {
+        this.enqueueJobAfterFailure(queue, job)
+      }
+    })
+    queue.on('failed', (job, err) => {
+      this.logError(
+        `Queue Job Failed - ID ${job?.id} - Error ${err}. Queuing another job...`
+      )
+      this.enqueueJobAfterFailure(queue, job)
+    })
+  }
+
+  /**
+   * Enqueues a job that processes the next slice of users after the slice
+   * that the previous job sucessfully processed.
+   * @param queue the StateMonitoringQueue to enqueue a job
+   * @param successfulJobData the jobData of the previous job that succeeded
+   * @param successfulJobResult the result of the previous job that succeeded
+   */
+  enqueueJobAfterSuccess(queue, successfulJobData, successfulJobResult) {
+    const {
+      data: { discoveryNodeEndpoint, moduloBase, currentModuloSlice }
+    } = successfulJobData
+    const { lastProcessedUserId } = successfulJobResult
+
+    queue.add({
+      startTime: Date.now(),
+      lastProcessedUserId,
+      discoveryNodeEndpoint,
+      moduloBase,
+      currentModuloSlice: (currentModuloSlice + 1) % moduloBase
+    })
+  }
+
+  /**
+   * Enqueues a job that picks up where the previous failed job left off.
+   * @param queue the StateMonitoringQueue to enqueue a job
+   * @param failedJob the jobData for the previous job that failed
+   */
+  enqueueJobAfterFailure(queue, failedJob) {
+    const {
+      data: {
+        lastProcessedUserId,
+        discoveryNodeEndpoint,
+        moduloBase,
+        currentModuloSlice
+      }
+    } = failedJob
+
+    queue.add({
+      startTime: Date.now(),
+      lastProcessedUserId,
+      discoveryNodeEndpoint,
+      moduloBase,
+      currentModuloSlice: (currentModuloSlice + 1) % moduloBase
+    })
+  }
+
+  /**
+   * Registers the logic that gets executed to process each new job from the queue.
+   * @param {*} queue the StateMonitoringQueue to consume jobs from
+   */
+  registerQueueConsumer(queue) {
```

Comment:
idk, this is a bit confusing - not what i would think of for queueConsumer, more like registerQueueJobProcessor

again, not sold here - open to your input

Reply:
renamed to registerQueueJobProcessor